### ISSUE TITLE
unable to overfit on 1 image

### ISSUE THREAD

**Issue by omerbrandis** [dbolya/yolact#598]
Hello , 

I have tried training on one image from my custom dataset.

my intent was to overfit so that when I evaluate on the same image i'll get an exact copy of the training annotation.

I was not able to do this, even though the reported loss was nearly zero for all metrics:
[10000]   10000 || B: 0.011 | C: 0.000 | M: 0.049 | S: 0.017 | T: 0.076 || ETA: 60 days, 11:14:29 || timer: 0.780
[20000]   20000 || B: 0.001 | C: 0.000 | M: 0.002 | S: 0.001 | T: 0.005 || ETA: 60 days, 10:29:48 || timer: 0.766
[30000]   30000 || B: 0.001 | C: 0.000 | M: 0.001 | S: 0.000 | T: 0.002 || ETA: 60 days, 7:38:31 || timer: 0.774
[40000]   40000 || B: 0.000 | C: 0.000 | M: 0.000 | S: 0.000 | T: 0.001 || ETA: 60 days, 3:27:07 || timer: 0.798
[50000]   50000 || B: 0.000 | C: 0.000 | M: 0.000 | S: 0.000 | T: 0.001 || ETA: 60 days, 3:53:50 || timer: 0.781

it looks like some logic was learned , hopefully ruling out problems with the annotation data.

eval output for each of the reported stages ( 10k to 50k)
![output1](https://user-images.githubusercontent.com/3157363/105819050-55a26e00-5fc0-11eb-915d-f51295cdc9ce.jpg)
![output2](https://user-images.githubusercontent.com/3157363/105819061-589d5e80-5fc0-11eb-9698-692f70ad3131.jpg)
![output3](https://user-images.githubusercontent.com/3157363/105819323-b3cf5100-5fc0-11eb-9f40-5b8ac5b33dcf.jpg)
![output4](https://user-images.githubusercontent.com/3157363/105819082-60f59980-5fc0-11eb-98e7-9fda20fa9334.jpg)
![output5](https://user-images.githubusercontent.com/3157363/105819093-63f08a00-5fc0-11eb-9b05-9f5ab71012ed.jpg)

very little difference between output2 and output5 - matching the loss report.

my questions:
1. is my intent valid ? (should I have been able to do this ?)
2. (provided the answer to "1" is yes)  any thoughts to why i was not successful ?

my config:
ffr1_dataset = dataset_base.copy({
    'name': 'ffr1 Dataset',

    'train_images':...
    'train_info':   ...

    'valid_images': ...
    'valid_info':   ...

    'has_gt': True,
    'class_names': ('apple','branch' , 'trellis')
})

ffr_base_config = yolact_base_config.copy({
    'name': 'ffr_base',

    # Dataset stuff
    'dataset': ffr1_dataset,
    'num_classes': len(ffr1_dataset.class_names) + 1,

    #
    'backbone': resnet50_dcnv2_backbone.copy({
        'selected_layers': list(range(1, 4)),

        'pred_aspect_ratios': [ [[1, 1/2, 2]] ]*5,
        'pred_scales': [[i * 2 ** (j / 3.0) for j in range(3)] for i in [24, 48, 96, 192, 384]],
        'use_pixel_scales': True,
        'preapply_sqrt': False,
        'use_square_anchors': False,
    }),

})

original image:
![img2784](https://user-images.githubusercontent.com/3157363/105819467-e8430d00-5fc0-11eb-908f-b568d65db0c0.jpg)

for anotation i've used the [coco-annotator](https://github.com/jsbroks/coco-annotator)
here's the file ( had to zip in order to upload) 
[img2784.json.zip](https://github.com/dbolya/yolact/files/5872077/img2784.json.zip)

Thanks,
Omer

**Comment by omerbrandis** [comment]
hi

increased top_k to 20 ( there are 14 annotated objects in train data)
i now see more info, but still very far off the original.

10k steps:
![10k_top20](https://user-images.githubusercontent.com/3157363/105843721-bfcb0b00-5fe0-11eb-8667-7546ed92a1f9.jpg)

50k steps:
![50k_top20](https://user-images.githubusercontent.com/3157363/105843736-c5285580-5fe0-11eb-8707-a9f39cdea9c8.jpg)

**Comment by sdimantsd** [comment]
Can you check if your annotations are valid?

**Comment by omerbrandis** [comment]
hello sdimantsd,

Can you please suggest a method for doing that ? ( how to check if annotations are valid)

thanks,
Omer.

**Comment by sdimantsd** [comment]
You can annotate them on the image.
I don't know any tool for doing it, so you can build your own or google for one.

**Comment by omerbrandis** [comment]
hello sdimandtsd,

I've created my annotations using [coco-annotator.](https://github.com/jsbroks/coco-annotator)
I can't say with 100% of certainty that it does not have any bugs,
but it not very likely that my use case is a fringe one and has not been tested.

you mentioned "testing the validity of the annotations", can you please be more specific as to what should be checked?

(+ is it not the responsibility of the software that receives input (=yolact) to check the inputs validity ?) )

thanks,
Omer.

**Comment by sdimantsd** [comment]
What I meant was for you to check that the polygons were correct.
It's true that yolact should check that it's true. But I do not know whether they carried out such a test or not.

**Comment by ynma-hanvo** [comment]
how many  images you created for training,   and is the mAP values good?

**Comment by omerbrandis** [comment]
Hi,

In this case, i'm using 1 image for training, and evaluating on same image. the intent is to overfit and make sure i can make the cnn learn the image by heart.
:-)
omer.

**Comment by ynma-hanvo** [comment]
during the training, there is a validation phase which shows   mAP ,  if you use the same validation data as the training data, you can check if mAP is growing up and stop at a good time.

**Comment by omerbrandis** [comment]
Hello ynma-hanvo,

I'm afraid i don't understand your answer.
1. why isn't loss the correct metric for this case (considering my eval data = training data)?

2. if using map, what is a good time/value to stop ?

thanks,
Omer.

**Comment by sdimantsd** [comment]
Hi @omerbrandis,
Did you solved this problem?
